example/advanced/server-io/advanced_server_io.cpp

```diff
@@ -0,0 +1,914 @@
+//
+// Copyright (c) 2016-2017 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+//------------------------------------------------------------------------------
+//
+// Example: Advanced server, one io_context per thread
+//
+//------------------------------------------------------------------------------
+
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/websocket.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/signal_set.hpp>
+#include <boost/asio/steady_timer.hpp>
+#include <boost/make_unique.hpp>
+#include <boost/bind.hpp>
+#include <boost/config.hpp>
+#include <algorithm>
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+using tcp = boost::asio::ip::tcp;               // from <boost/asio/ip/tcp.hpp>
+namespace http = boost::beast::http;            // from <boost/beast/http.hpp>
+namespace websocket = boost::beast::websocket;  // from <boost/beast/websocket.hpp>
+
+class round_robin
```

Comment:
Something to consider for the future, not important now:
```
template<class SelectionPolicy = round_robin>
class multi_io_context;
```

Reply:
That would be neat. Having a `least_connected` selection policy would be interesting. Give each io_context a weight based on the current number of sockets it's being used with, then next_context would return the io_context with the lowest weight.

Reply:
Hmm good idea! That requires counting the sockets but we could build some RAII counter thingy that you put in your session object. Here's another idea, in the io_context thread we call `run_one` instead of `run`, in a loop, and we can count I/O and compare how much time each thread is spending as a cheap proxy for utilization.

Reply:
Note that `run_one` may execute more than one CompletionHandler if the `io_context`'s Executor is wrapped with another Executor, e.g. `boost::asio::strand`.